Common/include/option_structure.inl

```diff
@@ -0,0 +1,2148 @@
+      this->marker = NULL;
+      this->var1 = NULL;
+      this->var2 = NULL;
+      this->flowdir = NULL;
+      this->field = NULL;
+      return newstring;
+    }
+
+    unsigned short nVals = totalVals / 7;
+    this->size = nVals;
+    this->marker = new string[nVals];
+    this->var1 = new su2double[nVals];
+    this->var2 = new su2double[nVals];
+    this->flowdir = new su2double*[nVals];
+    this->field = new unsigned short[nVals];
+
+    for (unsigned long i = 0; i < nVals; i++) {
+      this->flowdir[i] = new su2double[3];
+    }
+
+    for (unsigned long i = 0; i < nVals; i++) {
+      this->marker[i].assign(option_value[7*i]);
+        // Check to see if the enum value is in the map
+    if (this->m.find(option_value[7*i + 1]) == m.end()) {
+      string str;
+      str.append(this->name);
+      str.append(": invalid option value ");
+      str.append(option_value[0]);
+      str.append(". Check current SU2 options in config_template.cfg.");
+      return str;
+    }
+      Tenum val = this->m[option_value[7*i + 1]];
+      this->field[i] = val;
+
+      istringstream ss_1st(option_value[7*i + 2]);
+      if (!(ss_1st >> this->var1[i])) {
+        return badValue(option_value, "Riemann", this->name);
+      }
+      istringstream ss_2nd(option_value[7*i + 3]);
+      if (!(ss_2nd >> this->var2[i])) {
+        return badValue(option_value, "Riemann", this->name);
+      }
+      istringstream ss_3rd(option_value[7*i + 4]);
+      if (!(ss_3rd >> this->flowdir[i][0])) {
+        return badValue(option_value, "Riemann", this->name);
+      }
+      istringstream ss_4th(option_value[7*i + 5]);
+      if (!(ss_4th >> this->flowdir[i][1])) {
+        return badValue(option_value, "Riemann", this->name);
+      }
+      istringstream ss_5th(option_value[7*i + 6]);
+      if (!(ss_5th >> this->flowdir[i][2])) {
+        return badValue(option_value, "Riemann", this->name);
+      }
+    }
+
+    return "";
+  }
+
+  void SetDefault() {
+    this->marker = NULL;
+    this->var1 = NULL;
+    this->var2 = NULL;
+    this->flowdir = NULL;
+    this->size = 0; // There is no default value for list
+  }
+};
+
+template <class Tenum>
+class COptionGiles : public COptionBase{
+
+  map<string, Tenum> m;
+  unsigned short & size;
+  string * & marker;
+  unsigned short* & field; // Reference to the fieldname
+  string name; // identifier for the option
+  su2double * & var1;
+  su2double * & var2;
+  su2double ** & flowdir;
+  su2double * & relfac1;
+  su2double * & relfac2;
+
+public:
+  COptionGiles(string option_field_name, unsigned short & nMarker_Giles, string* & Marker_Giles, unsigned short* & option_field, const map<string, Tenum> m, su2double* & var1, su2double* & var2, su2double** & FlowDir, su2double* & relfac1, su2double* & relfac2) : size(nMarker_Giles),
+  	  	  	  	  marker(Marker_Giles), field(option_field), var1(var1), var2(var2), flowdir(FlowDir), relfac1(relfac1), relfac2(relfac2) {
+    this->name = option_field_name;
```

Comment:
I hope it is visible in my review that Codefactor found a tab here (and somewhere else). So shortly remove tabs with spaces and whilst you're at strip trailing whitespaces (but I guess you already did that second part)

Reply:
I missed the tab expansion here and in config.cpp, spaces should be fine.